src/vaadin-date-time-picker.html

```diff
@@ -476,6 +481,7 @@
             this.datePlaceholder = datePicker.placeholder;
             this.initialPosition = datePicker.initialPosition;
             this.showWeekNumbers = datePicker.showWeekNumbers;
+            this.invalid = datePicker.invalid;
```

Comment:
I think the `invalid` state should not be synced in this section ("Synchronize properties from slotted") on both date picker and time picker. It might cause issues when working on the Java version which uses the slotted elements.

Same goes for the `disabled` and `readonly` below which were added by me, but I noticed later that I think it doesn't make sense. These invalid, disabled and readonly should probably only be syncronized from DTP into the default/slotted so these could be moved down outside this if-else block.

It's fine if you just handle the `invalid` (in both `__datePickerChanged` and `__timePickerChanged`) which is related to this PR but if you want you can also move the other 2.

Reply:
Updated. Removed sync and tests for `disabled` and `readonly` as well.

Reply:
I meant that the `invalid`, `disabled` and `readonly` sync should (I think) be moved outside the if-else block to the end of the method (not completely removed) to the same place where `required` is now synchronized.

Now as you changed these they are (initially) synchronized only in case they're not slotted.

But actually I'm not sure about how this needs to work (with slotted) when using the Java API so I guess it's fine to leave these 3 properties as is. We can change this later if we notice some initial state doesn't work right when setup from Java.